Source/Core/Core/HW/DSPHLE/UCodes/AX.cpp

```diff
@@ -508,6 +507,50 @@ void AXUCode::SetMainLR(u32 src_addr)
   }
 }
 
+void AXUCode::RunCompressor(u16 threshold, u16 release_frames, u32 table_addr, u32 millis)
+{
+  // check for L/R samples exceeding the threshold
+  bool triggered = false;
+  for (u32 i = 0; i < 32 * millis; ++i)
+  {
+    if (std::abs(m_samples_left[i]) > int(threshold) ||
+        std::abs(m_samples_right[i]) > int(threshold))
+    {
+      triggered = true;
+      break;
+    }
+  }
+
+  const u32 frame_byte_size = 32 * millis * sizeof(s16);
+  u32 table_offset = 0;
+  if (triggered)
+  {
+    // one attack frame based on previous frame
+    table_offset = m_compressor_pos * frame_byte_size;
+    // next frame will start release
+    m_compressor_pos = release_frames;
+  }
+  else if (m_compressor_pos)
+  {
+    // I'm a bit confused that they do "dec $ACC0; sr @pos, $AC0.M".
+    // Seems like it should have been decm instead of dec? Very strange.
+    --m_compressor_pos;
```

Comment:
That seems weird, it would decrement a part of the accumulator that isn't used in the store. And also means that we _shouldn't_ decrement here if it is indeed correct.

Maybe theres a bug in DSPSpy (or whatever you used for disassembly of the UCode)?

Reply:
Mystery solved: The reason it works anyway is that the lower bits are zero, so e.g. `0x00'1234'0000 - 1 = 0x00'1233'ffff`.